ReSharper.FSharp/src/FSharp.Psi.Intentions/src/QuickFixes/AddSetterFix.fs

```diff
@@ -0,0 +1,47 @@
+namespace JetBrains.ReSharper.Plugins.FSharp.Psi.Features.Daemon.QuickFixes
+
+open JetBrains.ReSharper.Plugins.FSharp.Psi
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Features.Daemon.Highlightings
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Impl.Tree
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Tree
+open JetBrains.ReSharper.Psi.ExtensionsAPI
+open JetBrains.ReSharper.Resources.Shell
+
+type AddSetterFix(error: PropertyCannotBeSetError) =
+    inherit FSharpQuickFixBase()
+
+    let refExpr = error.RefExpr
+    let mutable declaration: IAccessorsNamesClauseOwner = null
+
+    override this.IsAvailable _ =
+        isValid refExpr &&
+
+        let declaredElement = refExpr.Reference.Resolve().DeclaredElement
+        let decl = declaredElement.GetDeclarations() |> Seq.tryExactlyOne
```

Comment:
Please check `declaredElement` for `null`, just in case. We can't guarantee that we'll always be able to get a declared element after error recovery in the compiler.